### PR TITLE
refactor: create helper function RelayRecoveredSig inside peerman

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
High m_nodes_mutex lock contention during high load

## What was done?
this commit should have a few benefits:
1. previous logic using ForEachNode results in locking m_nodes_mutex, a highly contended RecursiveMutex, AND m_peer_mutex(in GetPeerRef)
2. prior also resulted in calling .find over the m_peer_map for each node. Basically old logic was (probably) O(n(nlogn) the new logic results in acquiring m_peer_mutex once and looping over the list of peers, (probably) O(n)
3. Moves networking logic out of llmq/ and into actual net_processing.cpp

## How Has This Been Tested?
Hasn't really yet; it builds, but I need to run tests / maybe deploy to testnet mn

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v22.1.0-devpr6423.09a483dc. The image should be on dockerhub soon.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2339,6 +2342,18 @@ void PeerManagerImpl::RelayTransaction(const uint256& txid)
...
+void PeerManagerImpl::RelayRecoveredSig(const uint256& sigHash)
+{
+    const CInv inv{MSG_QUORUM_RECOVERED_SIG, sigHash};
+    LOCK(m_peer_mutex);
+    for(auto& it : m_peer_map) {
+        Peer& peer = *it.second;
+        if (!peer.m_wants_recsigs) continue;
+
+        PushInv(peer, inv);
+    };
```
Comment (UdjinM6) [inline src/net_processing.cpp:2354]:
```suggestion
    for (const auto& [_, peer] : m_peer_map) {
        if (peer->m_wants_recsigs) {
            PushInv(*peer, inv);
        }
    }
```

**Comment by PastaPastaPasta** [comment]
Okay, tested on a testnet node; and appears to work as expected. 

Now, disclaimer, this is on 3 testnet nodes. One running rc.1, one running this PR, and one running 6422. It is a bit hard to compare between rc.1 and the PRs as the PRs are built using debug, so their "contended time" is going to be higher, even if actual contention should be lower.

The total percentage of contentions caused by m_nodes_mutex went from ~41% -> 26%.

It did result in a significantly increased contention over `m_peer_mutex` going from only around 4% -> 10%

But the "contended time" for m_nodes_mutex went from 896839 -> 316444 = 580,395 
while the "contended time" for m_peer_mutex went from 75043 -> 192275 = -117,232; so it seems the net "contended time" is down around ~80%

File: src/llmq/signing.cpp [hunk 2]
```diff
@@ -633,12 +633,7 @@ void CSigningManager::ProcessRecoveredSig(const std::shared_ptr<const CRecovered
     WITH_LOCK(cs_pending, pendingReconstructedRecoveredSigs.erase(recoveredSig->GetHash()));
 
     if (m_mn_activeman != nullptr) {
-        CInv inv(MSG_QUORUM_RECOVERED_SIG, recoveredSig->GetHash());
-        connman.ForEachNode([&](const CNode* pnode) {
```
Comment (knst) [inline src/llmq/signing.cpp:637]: oh this change is soo good! Look to this one: 6149288d4818b7e5700657755104b95cbb03306c

Maybe you can drop all usages of CConnman from llmq code?

Comment (PastaPastaPasta) [reply src/llmq/signing.cpp:637]: Aye! Awesome, didn't even realize I'd removed all usages

**Review by knst — APPROVED** [review]
utACK 86e92c376a71c5cf5b061c0e7c178990c5d61c9f

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v22.1.0-devpr6423.4668db60. The image should be on dockerhub soon.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v22.1.0-devpr6423.86e92c37. A new comment will be made when the image is pushed.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v22.1.0-devpr6423.86e92c37. The image should be on dockerhub soon.

File: src/llmq/signing.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/llmq/signing.cpp:637]: @knst see: https://github.com/dashpay/dash/pull/6425 one more bites the dust!

**Review by UdjinM6 — APPROVED** [review]
utACK 86e92c376a71c5cf5b061c0e7c178990c5d61c9f

**Pull request merged**